### ISSUE TITLE
Typescript definitions 

### ISSUE THREAD

**Issue by wouter-leistra** [usablica/intro.js#1561]
## Description
Can introjs come with its own official typescript definitions like @types/introjs. There is some externally maintained one going around but when new features are introduced those do not get updated, like when the Got it button hide option was introduced last fall. So we had to start to maintain our own version based on the external one, but it is kind of a messy exercise.

**Comment by afshinm** [comment]
Sounds good. I was planning on converting the codebase to TS. That would also solve this issue.

**Comment by afshinm** [comment]
Yep, sorry about that @s00d. Moving to TS is desirable and that's what I intend to do, but it requires a lot of manual testing. I'm currently working on improving our test coverage (Cypress in particular) which would make this transition easier for us. Thanks again for your patient. I'm hoping to have some updates soon.

**Comment by s00d** [comment]
> Yep, sorry about that @s00d. Moving to TS is desirable and that's what I intend to do, but it requires a lot of manual testing. I'm currently working on improving our test coverage (Cypress in particular) which would make this transition easier for us. Thanks again for your patient. I'm hoping to have some updates soon.

Hello, updated
